### PR TITLE
[image] comply with constraint file in base image 

### PR DESCRIPTION
**Pull request by aslonnie**
also uses 1.3 syntax and heredoc for multiline commands

### CODE REVIEW

File: docker/base-deps/Dockerfile [hunk 1]
```diff
@@ -13,82 +15,112 @@ ENV LANG=C.UTF-8
...
-        "google-oauth" \
-        "azure-cli-core==2.40.0" \
-        "azure-identity==1.10.0" \
-        "azure-mgmt-compute==23.1.0" \
-        "azure-mgmt-network==19.0.0" \
-        "azure-mgmt-resource==20.0.0" \
-        "msrestazure==0.6.4"; \
-    fi;)
+
+COPY requirements_compiled.txt ./
```
[the remark this answers is not in the document]
Comment (aslonnie) [reply docker/base-deps/Dockerfile]: ok, fixed it, and some other issues too. slightly cleaner now.

**Comment by aslonnie** [comment]
should be ready to go

**Comment by aslonnie** [comment]
friendly ping for review.

**Comment by can-anyscale** [comment]
how are we testing this change?

**Comment by can-anyscale** [comment]
yeah maybe run a python 3.12 and default python release tests

**Review by can-anyscale — APPROVED** [review]

**Comment by aslonnie** [comment]
microbenchmark tests are passing, including py312.

the only one failing is the serve gce one, and I will ignore that one.

merging now.

**Pull request merged**